pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4251,8 +4243,16 @@ private RestException topicNotFoundReason(TopicName topicName) {
                     == null ? "has no metadata" : "has zero partitions";
             return new RestException(Status.NOT_FOUND, String.format(
                     "Partitioned Topic not found: %s %s", topicName.toString(), topicErrorType));
-        } else if (!internalGetList(Optional.empty()).contains(topicName.toString())) {
-            return new RestException(Status.NOT_FOUND, "Topic partitions were not yet created");
+        } else {
+            try {
+                if (!internalGetListAsync(Optional.empty())
+                    .get(pulsar().getConfiguration().getMetadataStoreOperationTimeoutSeconds(), TimeUnit.SECONDS)
+                    .contains(topicName.toString())) {
+                    return new RestException(Status.NOT_FOUND, "Topic partitions were not yet created");
+                }
+            } catch (Exception e) {
+                // failed to get reason, so fall through coarse reason
```

Comment:
I think we should throw an error, or return `return new RestException(Status.NOT_FOUND, e.getMessage());`

You can look at other people's opinions.

Reply:
en...It seems that `e.getMessage` is not the real reason I think  @nodece , Please take a look also @Technoboy- @Jason918

Reply:
I think you can add a new sync method called `internalGetList` like `getPartitionedTopicMetadata`.

Reply:
In order to reduce the duplicated codes, you can do it like sync(() -> internalGetListAsync()